src/simplex.h

```diff
@@ -57,14 +128,65 @@ std::ostream& operator<<(std::ostream& stream, const Solution<U>& solution) {
 template<typename T>
 class Simplex {
 private:
+    bool isCorrectInput;
     std::vector<T> func;
     Matrix<T> A;
     std::vector<T> b;
     std::vector<size_t> basic_indices;
     std::vector<T> basic_coeffs;
 
-    void validate() const {
-        // TODO: implement
+    void validateOnStart() {
+        std::vector<size_t> checkedBaseVars;
+
+        for (size_t idx = 0; idx < A.get_columns(); idx++) {
+            std::vector<T> temp = A.get_column(idx);
+
+            bool takenOne = false;
+            bool isBaseVar = true;
+            for (T num : temp) {
```

Comment:
it is better to use `const T&` to not call the copy constructor of `T` (for very simple types of T it will be slower, but for fraction, I think, it's faster)